### PR TITLE
add pacemaker resources configuration

### PR DESCRIPTION
**Pull request by tomjelinek**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -162,6 +174,116 @@ string, default: `my-cluster`
...
+* `instance_attrs` (optional) - List of sets of the resource's instance
+  attributes. Currently, only one set is supported. The exact names and values
+  of attributes, as well as whether they are mandatory or not, depends on the
+  resource or stonith agent.
+* `meta_attrs` (optional) - List of sets of the resource's meta attributes.
+  Currently, only one set is supported.
+* `operations` (optional) - List of the resource's operations.
+  * `action` (mandatory) - Operation action as defined by pacemaker and the
+    resource or stonith agent.
+  * `attrs` (mandatory) - Operation options, at least one option must be
```
Comment (richm) [inline README.md:224]: Are the names of the attributes guaranteed to be valid Python identifiers?  That is, can you have something like
"@attr-name" or some other name that isn't a valid Python identifier?  If so, then you won't be able to use `attribute1_name: value` - you'll have to use a `list` of `name: attr_name` `value: attr_value` pairs like this:
```yaml

    instance_attrs:
      - attrs:
          - name: attribute1_name
            value: "attribute1_value"
          - name: attribute2_name
            value: "attribute2_value"
```

File: tasks/pcs-cib-resource-clone-pcs-0.10.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    # *_attrs[0] instead of looping over *_attrs
+    cmd: >
+      pcs -f {{ __ha_cluster_tempfile_cib_xml.path | quote }}
+      resource
+      {% if resource_clone.promotable | default(false) %}
+        promotable
+      {% else %}
+        clone
+      {% endif %}
+      '{{ resource_clone.resource_id | quote }}'
```
Comment (richm) [inline tasks/pcs-cib-resource-clone-pcs-0.10.yml:36]: I don't think you need the single quotes here if you are using `| quote`

**Comment by richm** [comment]
[citest bad]

File: tasks/pcs-cib-resource-clone-pcs-0.10.yml (hunk 2 above)
Comment (tomjelinek) [reply tasks/pcs-cib-resource-clone-pcs-0.10.yml]: Right, I fixed that. Thanks.

File: README.md (hunk 1 above)
Comment (tomjelinek) [reply README.md:232]: No, that cannot be guaranteed. I switched to the format you proposed. Thanks.

**Comment by richm** [comment]
lgtm - @nhosoi or @spetrosi ptal

File: tasks/create-and-push-cib.yml [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+- name: Configure cluster resources
+  include_tasks: "pcs-cib-resource-create-pcs-0.10.yml"
+  vars:
+    resource: "{{ item }}"
+    resource_is_stonith: "{{ item.agent.startswith('stonith:') }}"
+  loop: "{{ ha_cluster_resource_primitives }}"
+
+- name: Configure cluster resource groups
+  include_tasks: "pcs-cib-resource-group-pcs-0.10.yml"
```
Comment (nhosoi) [inline tasks/create-and-push-cib.yml:86]: You don't need double quotes when the value is not or does not contain a variable although is it very minor.

File: tasks/create-and-push-cib.yml [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+  environment:
+    CIB_file: "{{ __ha_cluster_tempfile_cib_xml.path }}"
+  check_mode: no
+  changed_when: not ansible_check_mode
+
+
+# Build the new CIB
+
+- name: Configure cluster resources
+  include_tasks: "pcs-cib-resource-create-pcs-0.10.yml"
```
Comment (nhosoi) [inline tasks/create-and-push-cib.yml:79]: You don't need double quotes when the value is not or does not contain a variable although is it very minor.

**Comment by nhosoi** [comment]
[citest bad]

**Comment by tomjelinek** [comment]
Fedora 32 is failing due to dependency error in resource-agents package. This is a known issue, I have notified resource-agents maintainer about it. Tests for the feature implemented by this pull requests passed, though.

RHEL-x and Fedora 34 is failing with 'ERROR: Inventory not properly set up'. It does not even run the role's tests. Not sure what I can do about that.

File: tasks/create-and-push-cib.yml (hunk 4 above)
Comment (tomjelinek) [reply tasks/create-and-push-cib.yml:79]: Ah, ok. I tend to put quotes everywhere. Do you want me to remove them? Perhaps I could do another commit removing them from all the places throughout the role where they are not needed, including examples in readme file.

File: README.md [hunk 5]
```diff
@@ -196,18 +330,100 @@ all:
 
 ## Example Playbook
```
Comment (spetrosi) [inline README.md:331]:
```suggestion
## Example Playbooks
```

File: README.md [hunk 6]
```diff
@@ -196,18 +330,100 @@ all:
...
 
-Minimalistic example to create a cluster running no resources:
+### Minimalistic example to create a cluster running no resources
```
Comment (spetrosi) [inline README.md:333]:
```suggestion
### Creating a cluster running no resources
```

File: README.md [hunk 7]
```diff
@@ -196,18 +330,100 @@ all:
...
+- hosts: node1 node2
+  vars:
+    ha_cluster_cluster_name: "my-new-cluster"
+    ha_cluster_hacluster_password: "password"
+
+  roles:
+    - linux-system-roles.ha_cluster
+```
+
+### Cluster with fencing and several resources
```
Comment (spetrosi) [inline README.md:344]:
```suggestion
### Creating a cluster with fencing and several resources
```

File: README.md [hunk 8]
```diff
@@ -196,18 +330,100 @@ all:
...
+                value: "1"
+      - resource_id: "cloned-group"
+        promotable: yes
 
   roles:
     - linux-system-roles.ha_cluster
 ```
 
-To purge all cluster configuration, run this:
+### To purge all cluster configuration, run this
```
Comment (spetrosi) [inline README.md:426]:
```suggestion
### Purging all cluster configuration
```

**Review by spetrosi — COMMENTED** [review]
Adding minor readme corrections for readability, other than that looks good.

File: README.md (hunk 5 above)
Comment (tomjelinek) [reply README.md]: Thanks

**Comment by richm** [comment]
> Fedora 32 is failing due to dependency error in resource-agents package. This is a known issue, I have notified resource-agents maintainer about it. Tests for the feature implemented by this pull requests passed, though.

We will be dropping Fedora 32 support soon, so I wouldn't worry about this.

> 
> RHEL-x and Fedora 34 is failing with 'ERROR: Inventory not properly set up'. It does not even run the role's tests. Not sure what I can do about that.

It is a known problem with the Fedora 34 and RHEL 9 cloud images, so don't worry about this.

File: tasks/create-and-push-cib.yml (hunk 4 above)
Comment (nhosoi) [reply tasks/create-and-push-cib.yml:79]: Great. Thanks!

Comment (tomjelinek) [reply tasks/create-and-push-cib.yml:79]: I've opened PR #16 for this.

**Pull request merged**